src/paas_charm/_gunicorn/charm.py

```diff
@@ -23,6 +25,60 @@ def _workload_config(self) -> WorkloadConfig:
             framework_name=self._framework_name, unit_name=self.unit.name
         )
 
+    def check_gevent_package(self) -> bool:
+        """Check that gevent is installed.
+
+        Returns:
+            True if gevent is installed.
+        """
+        pip_list_command = self._container.exec(
+            ["python3", "-c", "'import gevent;print(gevent.__version__)'"]
+        )
+        list_output = pip_list_command.wait_output()[0]
```

Comment:
it the python3 command fails, it will exit with an error code that is not zero, and this will raise. I think the unit tests also do not test that the return code is not 0.